profile/get.go

```diff
@@ -28,7 +28,7 @@ const (
 
 var getProfileSingleInflight singleflight.Group
 
-// GetProfile fetches a profile. This function ensure that the profile loaded
+// GetProfile fetches a profile. This function ensures that the loaded profile
 // is shared among all callers. You must always supply both the scopedID and
 // linkedPath parameters whenever available.
 func GetProfile(source profileSource, id, linkedPath string) ( //nolint:gocognit
```

Comment:
IMHO the function documentation does not explain what happes if I don't have the `scopeID` or `linkedPath available`.
We may also rename the type of the first parameter and remove the `profile` prefix as this is the package name and stutters when (trying) to use from somewhere else: `profile.profileSource`

Reply:
I improved the documentation.
Yes, `profile.profileSource` stutters a bit, but just using `source` is not explicit enough, IMHO - I tried it.